### DISCUSSION TITLE
Twitter: Hide retweets of an author's own tweets

### DISCUSSION THREAD

**Discussion by v1gnesh** [uBlockOrigin/uAssets#23469] [Q&A] [answered]
### Prerequisites

- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367), [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) or [_a shortener/hosting site_](https://github.com/uBlockOrigin/uAssets/discussions/17361) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Search [here](https://github.com/uBlockOrigin/uAssets/issues) to find out.)
- [X] I did not remove any of the [default filter lists](https://github.com/uBlockOrigin/uAssets/assets/66517106/6e8d024a-5869-47cf-b142-b2357eb6c7d0), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable additional filter lists, or I have verified that the issue still occurs without enabling additional filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin along with other content blockers.
- [X] I have verified that the web browser's built-in content blocker/tracking protection, [network wide/DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking), or my VPN is not causing the issue.
- [X] I have verified that other extensions are not causing the issue.
- [X] If this is about a breakage or detection, I have verified that it is caused by uBlock Origin and isn't a site issue.
- [ ] I did not answer truthfully to **ALL** the above checkboxes.

### URL(s) where the issue occurs.

```yaml
Example: https://x.com/__mharrison__/status/1785702063965339685.
See how the author reposts/retweets his own tweet.
Would like to hide such self-reposts/retweets from my home timeline.
```

### Description

I'd like to hide some people retweeting their own stuff. Ideally limiting this to specific "handles" I can choose, instead of all retweets of the same author's post. 
An option to hide self-promotion basically.

### Other extensions used

No other content blockers.

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

![Screenshot from 2024-04-27 09-35-18](https://github.com/uBlockOrigin/uAssets/assets/234149/0570bdf9-6bc7-4f81-9866-46f0d3249ff2)

</details>

### Configuration

<details>
<summary>Details</summary>

```yaml


```
</details>

**Comment by u-RraaLL** [comment]
Provide an example URL.

**Reply by v1gnesh** [reply]
Thank you @u-RraaLL, here's one - https://x.com/__mharrison__/status/1785702063965339685
See how the author reposts/retweets his own tweet.
Would like to hide such self-reposts/retweets from my home timeline.

**Reply by u-RraaLL** [reply, marked as answer]
twitter.com,x.com##[aria-label="Timeline: Your Home Timeline"] [data-testid="cellInnerDiv"]:has([href="/__mharrison__"]>[data-testid="socialContext"]):has([data-testid="User-Name"] [href="/__mharrison__"])

**Reply by v1gnesh** [reply]
Thank you, is there a way to just check if both `href`s are equal, without specifying one username, i.e., generalize it?

**Reply by u-RraaLL** [reply]
Possibly with xpath, but I'm not familiar with it enough to make it work.

**Reply by Yuki2718** [reply]
I don't think it's possible. Would be possible if regex in procedural filters supported capturing and share the capture among other operators, but this won't be a trivial amount of work.

**Reply by u-RraaLL** [reply]
https://stackoverflow.com/questions/25914462/compare-attribute-values-using-xpath

This kinda sounds like it should be possible, but I wasn't able to craft a working expression.

**Reply by Yuki2718** [reply]
Okay, great, it's a brain exercise. I came up with `twitter.com##:xpath(//a[@id and @href=ancestor::article[@data-testid="tweet"]//div[@data-testid="User-Name"]/div/div/a[1]/@href]/span[@data-testid="socialContext"]/ancestor::div[@data-testid="cellInnerDiv"])`

**Reply by u-RraaLL** [reply]
Nice. It makes me wonder why my attempt failed though. It was a very similar logic only I started from the retweet icon so it had more steps.

Can you make it work for ebay? https://github.com/uBlockOrigin/uAssets/issues/23491#issuecomment-2087461412

**Comment by YUUUCC** [comment]
Is there a way to hide all retweet/repost context in the Following timeline?

**Reply by u-RraaLL** [reply]
Just Following and not the whole homepage?

**Reply by YUUUCC** [reply]
Yes, just the Following.
